misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormatterUtils.java

```diff
@@ -551,6 +554,18 @@ static boolean addNewTrailingLine(NonTerminalNode parent, NonTerminalNode node)
         return true;
     }
 
+    /**
+     * Converts the syntax tree into source code, remove superfluous spaces and newlines at the ending and returns it
+     * as a string.
+     *
+     * @param syntaxTree       syntaxTree
+     * @return source code as a string
+     */
+    public static String toFormattedSourceCode(SyntaxTree syntaxTree) {
+        return syntaxTree.toSourceCode().trim() +
+                System.getProperty(LINE_SEPARATOR);
```

Comment:
There are multiple places we call `System.getProperty(LINE_SEPARATOR)`, instead of keeping just the `LINE_SEPARATOR` as a static field, let's keep the actual return value.